src/Compilers/CSharp/Test/Symbol/Symbols/StaticAbstractMembersInInterfacesTests.cs

```diff
@@ -10946,5 +10966,1580 @@ unsafe class Test
+            // In metadata, only explicit implementations are considered
+
+            var ilSource = @"
+.class interface public auto ansi abstract I1
+{
+    .method public hidebysig static abstract virtual 
+        void M01 () cil managed 
+    {
+    } // end of method I1::M01
+} // end of class I1
+
+.class public auto ansi beforefieldinit C1
+    extends System.Object
+    implements I1
+{
+    .method private hidebysig  
+        static void I1.M01 () cil managed 
+    {
+        .override method void I1::M01()
+        .maxstack 8
+
+        IL_0000: ret
+    } // end of method C1::I1.M01
+
+    .method public hidebysig static 
+        void M01 () cil managed 
+    {
+        IL_0000: ret
+    } // end of method C1::M01
+
+    .method public hidebysig specialname rtspecialname 
+        instance void .ctor () cil managed 
+    {
+        IL_0000: ldarg.0
+        IL_0001: call instance void System.Object::.ctor()
+        IL_0006: ret
+    } // end of method C1::.ctor
+} // end of class C1
+
+.class public auto ansi beforefieldinit C2
+    extends C1
+    implements I1
+{
+    .method public hidebysig static 
+        void M01 () cil managed 
+    {
+        IL_0000: ret
+    } // end of method C2::M01
+
+    .method public hidebysig specialname rtspecialname 
+        instance void .ctor () cil managed 
+    {
+        IL_0000: ldarg.0
+        IL_0001: call instance void C1::.ctor()
+        IL_0006: ret
+    } // end of method C2::.ctor
+} // end of class C2
+";
+            var source1 =
+@"
+public class C3 : C2
+{
+}
+
+public class C4 : C1, I1
+{
+}
+
+public class C5 : C2, I1
+{
+}
+";
+
+            var compilation1 = CreateCompilationWithIL(source1, ilSource, options: TestOptions.DebugDll,
+                                                 parseOptions: TestOptions.RegularPreview,
+                                                 targetFramework: TargetFramework.NetCoreApp);
+
+            var c1 = compilation1.GlobalNamespace.GetTypeMember("C1");
+            var m01 = c1.Interfaces().Single().GetMembers().OfType<MethodSymbol>().Single();
+
+            var c1M01 = (MethodSymbol)c1.FindImplementationForInterfaceMember(m01);
+
+            Assert.Equal("void C1.I1.M01()", c1M01.ToTestDisplayString());
+            Assert.Same(m01, c1M01.ExplicitInterfaceImplementations.Single());
+
+            var c2 = compilation1.GlobalNamespace.GetTypeMember("C2");
+            Assert.Same(c1M01, c2.FindImplementationForInterfaceMember(m01));
+
+            var c3 = compilation1.GlobalNamespace.GetTypeMember("C3");
+            Assert.Same(c1M01, c3.FindImplementationForInterfaceMember(m01));
+
+            var c4 = compilation1.GlobalNamespace.GetTypeMember("C4");
+            Assert.Same(c1M01, c4.FindImplementationForInterfaceMember(m01));
+
+            var c5 = compilation1.GlobalNamespace.GetTypeMember("C5");
+
+            Assert.Equal("void C2.M01()", c5.FindImplementationForInterfaceMember(m01).ToTestDisplayString());
```

Comment:
Is this a scenario that requires both a synthesized "forwarding" method to be added to C5 as well as a MethodImpl entry? Should we verify that such a forwarding method was produced?

Reply:
>Is this a scenario that requires both a synthesized "forwarding" method to be added to C5 as well as a MethodImpl entry? Should we verify that such a forwarding method was produced?

I believe other tests sufficiently cover this aspect. The primary goal of this test is to see what method we consider to implement the interface.